app/actions/app_apply_manifest.rb

```diff
@@ -188,21 +193,15 @@ def create_binding_message(service_instance_guid, app_guid, manifest_service_bin
       )
     end
 
-    def raise_binding_error!(service_instance, message)
-      error_message = "For service '#{service_instance.name}': #{message}"
-      raise ServiceBindingError.new(error_message)
-    end
-
-    def binding_exists?(service_instance, app)
-      binding = ServiceBinding.first(service_instance: service_instance, app: app)
-      binding && !binding.create_failed?
-    end
-
-    def binding_being_deleted!(service_instance, app)
-      binding = ServiceBinding.first(service_instance: service_instance, app: app)
-      if binding && binding.operation_in_progress? && binding.last_operation.type == 'delete'
-        raise_binding_error!(service_instance, 'An existing binding is being deleted. Try recreating the binding later.')
+    def raise_invalid_binding_error!(service_instance, binding)
```

Comment:
This could then be split up into:
- `def binding_operation_in_progress!(service_instance, operation_type)`
- `def binding_deletion_failed!(service_instance)`

Both would also use the `raise_binding_error!` method.